command/exclude.go

```diff
@@ -0,0 +1,55 @@
+package command
+
+import (
+	"os"
+	"path"
+
+	"github.com/peak/s5cmd/storage/url"
+)
+
+// IsURLExcluded checks whether given urlPath matches any of the exclude patterns.
+func isURLExcluded(srcurl *url.URL, excludeUrls []*url.URL) bool {
+	if len(excludeUrls) == 0 {
+		return false
+	}
+
+	for _, excludeUrl := range excludeUrls {
+		if excludeUrl.Match(srcurl.Path) {
+			return true
+		}
+	}
+	return false
+}
+
+func createExcludeUrls(excludes []string, srcurls ...*url.URL) ([]*url.URL, error) {
+	result := make([]*url.URL, 0)
+	for _, srcurl := range srcurls {
+		for _, exclude := range excludes {
+			if exclude == "" {
+				continue
+			}
+			sourcePrefix := srcurl.GetUntilPrefix()
+			excludeStringUrl := sourcePrefix + exclude
+			if !srcurl.IsRemote() {
+				// Stat call is needed because we need to  understand
+				// if given local source is a directory,
+				// if it is, then we need to join the path.
+				// the given local source might be a directory, but it may
+				// not have / or \ as suffix.
+				obj, err := os.Stat(srcurl.Absolute())
```

Comment:
I think we should use `storage.Storage.Stat()` instead of using `os.Stat()`

Reply:
I changed the logic of the exclude feature. Rather than using URL package, I check if the object matches the given exclude pattern and I believe this is simpler approach by the following reasons.
1. Creating URLs for exclude patterns is a complicated process. We need to check if the source is remote or local, and behave differently.
2. If the source is local, then we need to check if it is a folder to avoid the complications of adding exclude pattern to given directory name. For example if source is given such as 
```
s5cmd cp --exclude "*.txt" directory s3://bucket
```
then we need to create URLs with `directory/*.txt`. Therefore, we need to check if given local source is a folder, and if it is, we need to join source and exclude URL. If we do not join, then it will be `directory*.txt`. However this does not solve all the issues. Because user can give the source like this. 

```
s5cmd cp --exclude "*.txt" directory/* s3://bucket
```
Then we need to check if `directory/*` is a folder and we know that it is not a folder, therefore it gives an error. This is one of the possible complications which may occur using URL package.

Therefore I believe that, it is not necessary to create URLs to exclude objects by using `URL.Match` method. We can use similar and simpler approach which is implemented in the other [PR](https://github.com/peak/s5cmd/pull/334).